src/interactive-window/editor-integration/codeGenerator.ts

```diff
@@ -0,0 +1,292 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+'use strict';
+import * as hashjs from 'hash.js';
+import {
+    Disposable,
+    NotebookCell,
+    NotebookCellKind,
+    Position,
+    Range,
+    TextDocumentChangeEvent,
+    TextDocumentContentChangeEvent
+} from 'vscode';
+
+import { splitMultilineString } from '../../webviews/webview-side/common';
+import { IDocumentManager } from '../../platform/common/application/types';
+import { traceInfo } from '../../platform/logging';
+import { IConfigurationService } from '../../platform/common/types';
+import { getCellResource, uncommentMagicCommands } from './cellFactory';
+import { CellMatcher } from './cellMatcher';
+import { ICellHash, IInteractiveWindowCodeGenerator, IGeneratedCodeStore } from './types';
+import { getInteractiveCellMetadata } from '../helpers';
+
+// This class provides hashes for debugging jupyter cells. Call getHashes just before starting debugging to compute all of the
+// hashes for cells.
+export class CodeGenerator implements IInteractiveWindowCodeGenerator {
```

Comment:
This is the old `CellHashProvider`, now responsible for just generating the code and storing it in the storage.

Reply:
Might make sure to tweak the class level comments before final review.

Reply:
Done